File: backend-dotnetcore/ApiModels/ActorSummary.cs
using Microsoft.AspNetCore.Mvc.ModelBinding;
using Swashbuckle.AspNetCore.Annotations;

namespace RUG.WebEng.Movies.ApiModels;

public class ActorSummary
{
    [BindNever]
    [SwaggerSchema(ReadOnly = true)]
    public int Id { get; set; }
    public string Name { get; set; }

    public static ActorSummary FromDatabase(Models.Actor actor) =>
        new()
        {
            Id = actor.Id,
            Name = actor.Name
        };
}

File: backend-express/api-models/actor-summary.ts
import { Actor as DbActor } from "models/actor";

export class ActorSummary {
    id: number;
    name: string;

    public static fromDatabase(actor : DbActor): ActorSummary {
        return {
            id: actor.id,
            name: actor.name
        };
    }
}